client/src/models/Task.js

```diff
@@ -153,11 +178,75 @@ export default class Task extends Model {
     super(Model.fillObject(props, Task.yupSchema))
   }
 
+  isTopLevelTask() {
+    return _isEmpty(this.customFieldRef1)
+  }
+
+  fieldSettings() {
+    return this.isTopLevelTask()
+      ? Settings.fields.task.topLevel
+      : Settings.fields.task.subLevel
+  }
+
+  periodAssessmentYupSchema() {
+    return this.isTopLevelTask()
+      ? Task.topLevelAssessmentCustomFieldsSchema
+      : Task.subLevelAssessmentCustomFieldsSchema
+  }
+
+  periodAssessmentConfig() {
+    return this.fieldSettings().assessment?.customFields
+  }
+
   iconUrl() {
     return TASKS_ICON
   }
 
   toString() {
     return `${this.shortName}`
   }
+
+  getAssessmentResults(dateRange) {
+    const publishedReportsUuids = this.publishedReports.map(r => r.uuid)
+    const taskAssessmentNotes = this.notes
+      .filter(
+        n =>
+          n.type === NOTE_TYPE.ASSESSMENT &&
+          n.noteRelatedObjects.length === 2 &&
+          n.noteRelatedObjects.filter(
+            ro =>
+              ro.relatedObjectType === Report.relatedObjectType &&
+              publishedReportsUuids.includes(ro.relatedObjectUuid)
+          ).length &&
+          (!dateRange ||
+            (n.createdAt <= dateRange.end && n.createdAt >= dateRange.start))
+      )
+      .map(ta => JSON.parse(ta.text))
+    const assessmentResults = {}
+    taskAssessmentNotes.forEach(o =>
+      Object.keys(o).forEach(k => {
+        assessmentResults[k] = assessmentResults[k] || []
```

Comment:
I understand what this code does, but it smells a bit too clever to me (such as that it does a useless assignment once `assessmentResults[k]` has been initialised).

Reply:
Done